src/wasm.h

```diff
@@ -90,6 +90,9 @@ struct Name : public cashew::IString {
 
 // An index in a wasm module
 typedef uint32_t Index;
```

Comment:
@kripken Assert added in toIndex(). I'm requesting a merge now. After merge, I'll go back and look over the additional suggestions. Changes are plentiful enough that I don't want to complicate the PR further.